mobile/src/employee-details/employee-details.tsx

```diff
@@ -15,23 +17,43 @@ import { Employee } from '../reducers/organization/employee.model';
 import { ApplicationIcon } from '../override/application-icon';
 import { layoutStylesForEmployeeDetailsScreen } from './styles';
 import { openCompanyAction } from './employee-details-dispatcher';
+import { loadCalendarEvents } from '../reducers/calendar/calendar.action';
+import { CalendarEvent } from '../reducers/calendar/calendar-event.model';
+import { eventDialogTextDateFormat } from '../calendar/event-dialog/event-dialog-base';
+import { EmployeeDetailsEventsList } from './employee-details-events-list';
 
 interface EmployeeDetailsProps {
-    employee: Employee;
+    employee?: Employee;
     department: Department;
-    layoutStylesChevronPlaceholder: ViewStyle;
-
+    layoutStylesChevronPlaceholder?: ViewStyle;
+    events?: Map<string, CalendarEvent[]>;
+    eventsPredicate?: (event: CalendarEvent) => boolean;
 }
+
+const mapStateToProps = (state: AppState, props: EmployeeDetailsProps): EmployeeDetailsProps => ({
+    department: props.department,
+    events: state.calendar.calendarEvents.events,
+    eventsPredicate: (event: CalendarEvent) => {
```

Comment:
New lambda will be returned and the state will be marked as _changed_

Reply:
addressed